### PR TITLE
Remove Maintainer rules about merging specifics

### PR DESCRIPTION
**Pull request by staltz**
I'm proposing to remove 2 rules that have been somewhat contentious in an SSB thread (`%9TGNIbIqh4MUk7zQAhHETld5ynebiimHE3pdJgU/vWg=.sha256`). 

I think line 70 ("*Maintainers SHOULD merge project pull requests in a fast and friendly manner.*") is a really good one that covers in a broad sense what the PR merging experience should look line, and the proposed removed lines are more specific and seem to bring disagreement.

To address each one:

> Maintainers SHOULD NOT suggest small improvements on project pull requests.

"Small improvements" could be interpreted in various ways, they could be anything between "remove code style changes that are inconsistent with the project" and "replace `const x = foo.x` with `const {x} = foo`" (arguably a nitpick). The fact that maintainers are not supposed to ask or suggest *any* small improvements is contentious and I think we will do better by trusting the maintainers that we chose, and keeping this README high-level, not specific. 

> Maintainers SHOULD suggest their improvements by opening their own pull request.

This one says "improvements", not "small improvements", giving the impression that maintainers should not suggest anything at all to a contributor's PR. 

More importantly, it assumes that maintainers also should not **add** commits to the PR. I think Git gives us many ways of solving issues such as adding commits to the PR, as well as squash-and-merge. @christianbundy expressed in SSB that he dislikes squash-and-merge, but I see it being often very useful for a friendly contribution workflow. I think this README should refrain from specific Git guidelines, and trust the maintainers to keep a friendly contribution environment, regardless of their preferred Git strategies.

### CODE REVIEW

**Review by arj03 — APPROVED** [review]

**Comment by christianbundy** [comment]
Thanks for proposing this patch! I've noticed that [lots of SSBC pull requests have lots of minor suggestions](https://github.com/pulls?q=is%3Apr+user%3Assbc+sort%3Acomments-desc), which I'd like to avoid in SSB-JS projects. It seems to me that the maintainer should be tasked with making the decision to identify a blocking concern or merge the patch, *not* suggest unnecessary changes. If the rule is unclear I'd enthusiastically approve a patch to make it more clear, but I'd prefer to keep some language outlining our commitment against unnecessary suggestions.

## Clarification

> "Small improvements" could be interpreted in various waysthey could be anything between "remove code style changes that are inconsistent with the project" and "replace const x = foo.x with const {x} = foo" (arguably a nitpick).

Yes! Maintainers are humans that we trust to interpret the rules and use their best judgment. If we can write a test for something (e.g. code style) then we should, but for things that we can't write tests for (`foo` being renamed to `eat_my_shorts`) we get to trust maintainers to block the patch from being merged. Maybe "unnecessary changes" would be more clear than "small improvements"? If a change is necessary for merging, then it [by definition] needs to change, but I think maintainers should avoid making suggestions that are optional.

> This one says "improvements", not "small improvements", giving the impression that maintainers should not suggest anything at all to a contributor's PR.

I'd approve a patch to improve this sentence if it's not clear enough. :heavy_check_mark:

## Opinion

This has been discussed a bit on SSB, so I'll just summarize my main points of disagreement:

- This is meant to communicate ideals to each other, contributors, and **future maintainers**.
  - I'd prefer to correct rules rather than delete them and move back toward [structurelessness](https://www.jofreeman.com/joreen/tyranny.htm).
- I think nitpicks (i.e. suggestions on non-blocking "problems") are net negative and exhausting.
  - I'd prefer that if a problem isn't big enough to block a merge, it shouldn't be brought up at all.
- Adding your commits to a PR you're reviewing is a circumvention of the review process.
  - I'd prefer to avoid having maintainers review and approve their own code or commits.
- Rewriting contributor commit history [causes problems and should usually be avoided](https://yarchive.net/comp/linux/git_rebase.html).
  - I'd prefer to avoid asking people to rewrite their history unless it's a blocking concern.
  - To be clear, the existing rule *does not* prevent maintainers from 'squash and merge' or rebase merging. It only prevents maintainers from nitpicking contributors about their code or commit history **unless there's a problem big enough to block the merge**.

With that said, I won't be bummed or anything if this is merged and I'm excited to continue the 'drunken stumble to greatness' by proposing patches and merging the ones that are enthusiastically approved. :heart:

**Comment by soapdog** [comment]
I am on the fence on this one and can't yet make my mind.

On one side, I like it the way it is because I have been hurt before with PRs that works just like _Brazilian Bureaucracy_. What I mean by that is that the maintainer keeps asking for small improvements in a cycle and the PR is never done and never merged regardless of how many alterations thus adding a lot of friction to contribution (this has happened in non-ssb projects, just to make sure no one is reading any subtext).

So I guess those guidelines are there to prevent this friction cycle where the submitter of the PR doesn't have a _clear picture of all the changes they need to make_ for the PR to be accepted. This in my mind doesn't preclude some back and forth to make sure the PR is in a good condition. Merging PRs shouldn't be a matter of they arriving in a perfect state. We can interact a bit on them but, we shouldn't stay in this _changes needed_ cycle forever.

All this text was necessary because I think that what we need is something that is between the current guideline and this proposed PR. We need some phrases there that enable maintainers to request changes while protecting contributors from patching like they are blind squirrels in a dungeon trying to please some inscrutable master.

**Comment by staltz** [comment]
After a lot of discussion, I'm starting to see what your/our opinions are. I will try to avoid talking about specifics (I'm not going down the git rebase rabbit hole and commit history for instance, I do have diverging opinions on that). I will try to summarize (because long discussions drain a lot more energy than short ones). In bold are key terms that make a lot of difference.

- I _think_ @christianbundy is saying "maintainers should not ask PR authors to do changes that are **non-blocking**".
- I _think_ @soapdog is saying that "maintainers should make it clear **upfront** what is blocking and what is non-blocking".

The two sentences above are things **I agree** on, so I think I can do another PR that attempts to rephrase to find a middle ground.

**Review by soapdog — APPROVED** [review]

**Comment by staltz** [comment]
#22 supercedes this PR

**Pull request closed without merging**
